apollo-router/src/plugins/file_uploads/mod.rs

```diff
@@ -0,0 +1,644 @@
+                                .build()?,
+                        ),
+                    })
+                }
+                .boxed()
+            })
+            .service(service)
+            .boxed()
+    }
+
+    fn execution_service(&self, service: execution::BoxService) -> execution::BoxService {
+        if !self.enabled {
+            return service;
+        }
+        ServiceBuilder::new()
+            .checkpoint(|req: execution::Request| {
+                let context = req.context.clone();
+                Ok(match execution_layer(req) {
+                    Ok(req) => ControlFlow::Continue(req),
+                    Err(err) => ControlFlow::Break(
+                        execution::Response::error_builder()
+                            .errors(vec![err.into()])
+                            .context(context)
+                            .build()?,
+                    ),
+                })
+            })
+            .service(service)
+            .boxed()
+    }
+
+    fn subgraph_service(
+        &self,
+        _subgraph_name: &str,
+        service: subgraph::BoxService,
+    ) -> subgraph::BoxService {
+        if !self.enabled {
+            return service;
+        }
+        ServiceBuilder::new()
+            .oneshot_checkpoint_async(|req: subgraph::Request| {
+                subgraph_layer(req)
+                    .boxed()
+                    .map(|req| Ok(ControlFlow::Continue(req)))
+                    .boxed()
+            })
+            .service(service)
+            .boxed()
+    }
+}
+
+fn get_multipart_mime(req: &router::Request) -> Option<MediaType> {
+    req.router_request
+        .headers()
+        .get(CONTENT_TYPE)
+        // Ignore parsing error, since they are reported by content_negotiation layer.
+        .and_then(|header| header.to_str().ok())
+        .and_then(|str| MediaType::parse(str).ok())
+        .filter(|mime| mime.ty == MULTIPART && mime.subty == FORM_DATA)
+}
+
+async fn router_layer(
+    req: router::Request,
+    limits: MultipartRequestLimits,
+) -> UploadResult<router::Request> {
+    if let Some(mime) = get_multipart_mime(&req) {
+        let boundary = mime
+            .get_param(BOUNDARY)
+            .ok_or_else(|| FileUploadError::InvalidMultipartRequest(multer::Error::NoBoundary))?
+            .to_string();
+
+        let (mut request_parts, request_body) = req.router_request.into_parts();
+
+        let mut multipart = MultipartRequest::new(request_body, boundary, limits);
+        let operations_stream = multipart.operations_field().await?;
+
+        req.context
+            .extensions()
+            .lock()
+            .insert(RouterLayerResult { multipart });
+
+        let content_type = operations_stream
+            .headers()
+            .get(CONTENT_TYPE)
+            .cloned()
+            .unwrap_or_else(|| HeaderValue::from_static("application/json"));
+        request_parts.headers.insert(CONTENT_TYPE, content_type);
```

Comment:
Just to make sure, do we want to `insert` or `append` ?

Reply:
The content type needs to replace the existing `multipart` content type, so I think it's insert. @IvanGoncharov Do you agree?

Reply:
Yes we need to replace it, otherwise we will see error in "content_negotition" layer since router don't know what to do with Multipart